osquery/tables/system/windows/registry.cpp

```diff
@@ -365,7 +374,7 @@ Status queryKey(const std::string& keyPath, QueryData& results) {
       case REG_FULL_RESOURCE_DESCRIPTOR:
       case REG_RESOURCE_LIST:
       case REG_BINARY:
-        for (size_t j = 0; j < cbMaxValueData; j++) {
+        for (size_t j = 0; j < lpData; j++) {
```

Comment:
[The internet](https://devblogs.microsoft.com/oldnewthing/20240214-00/?p=109400) suggests that the returned data size is an upper bound -- it includes headroom for an optional null, and whatever else may arise. So I think we need to iterate this buffer until either we hit the max size of `lpData` or a null byte

Reply:
Ah, but `lpData` is set again in the second call to the _actual_ length of the results returned. So I think this does work? See `pcbData` in https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-reggetvaluew.